crawler-cli/src/main/java/org/gbif/crawler/pipelines/dwca/DwCAToAvroCommandVerification.java

```diff
@@ -0,0 +1,144 @@
+package org.gbif.crawler.pipelines.dwca;
+
+import org.gbif.common.messaging.api.messages.DwcaValidationFinishedMessage;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.eclipse.jdt.internal.core.Assert;
+
+/**
+ * Verifies the configuration parameters like existence of input and output path, and enables with calculated configuration parameters helping DwCAToAvro Conversion
+ */
+public class DwCAToAvroCommandVerification {
+
+  private final DwCAToAvroConfiguration configuration;
+  private DwcaValidationFinishedMessage receivedMessage;
+
+  public static DwCAToAvroCommandVerification of(DwCAToAvroConfiguration configuration) {
+    return new DwCAToAvroCommandVerification(configuration);
+  }
+
+  private DwCAToAvroCommandVerification(DwCAToAvroConfiguration configuration) {
+    this.configuration = configuration;
+  }
+
+  public DwCAToAvroCommandVerification with(DwcaValidationFinishedMessage receivedMessage) {
+    this.receivedMessage = receivedMessage;
+    return this;
+  }
+
+  /**
+   * verifies parameters and return calculated configuration parameters
+   *
+   * @return Calculated Configuration Parameters
+   */
+  public DwCA2AvroConfigurationParameter verifyParametersAndGetResourceConfigurations() {
+    /*
+      checks existence of archiveRepository provided in the yaml config
+     */
+    String inputBasePath = configuration.archiveRepository.endsWith(File.separator)
+      ? configuration.archiveRepository
+      : configuration.archiveRepository + File.separator;
+    Assert.isLegal(new File(inputBasePath).exists(), "Illegal Argument archiveRepository is not a valid path");
+    /*
+      calculates and checks existence of DwC Archive
+    */
+    String absoluteDwCAExportPath = inputBasePath + receivedMessage.getDatasetUuid().toString();
+    String absoluteDwCAPath = absoluteDwCAExportPath + ".zip";
+    Assert.isLegal(new File(absoluteDwCAPath).exists(),
+                   "Illegal Argument " + absoluteDwCAPath + " Input DwC Archive not available");
+
+    Configuration config = new Configuration();
```

Comment:
you do not need to create a Configuration instance here, it can be done inside the getFileSystemFunction

Reply:
done.